### PR TITLE
#2360: fix bug where Groovy 3 internal generated methods were detected as test methods

### LINKED ISSUE #2360: Groovy 3 Unit Test Class: TestNGException at Parameters.java:437
7.3.0

### Expected behavior
Test pass without exception

### Actual behavior
Gradle suite > Gradle test > foo.FooTest > setMetaClass FAILED
    org.testng.TestNGException at Parameters.java:437
Received a failure event for test with unknown id '55.7'. Registered test ids: '[55.3, 55.2, 55.1, :test]'

### Is the issue reproductible on runner?

- [ ] Shell
- [ ] Maven
- [X] Gradle
- [ ] Ant
- [ ] Eclipse
- [X] IntelliJ
- [ ] NetBeans

Java version AdoptOpenJDK (build 25.262-b10, mixed mode)
Gradle version 6.6.1
Groovy version 3.0.5

Project attached: [g3tng.tar.gz](https://github.com/cbeust/testng/files/5145226/g3tng.tar.gz)

`src/test/groovy/foo/FooTest.groovy`
```
package foo
import groovy.transform.CompileStatic
import org.testng.annotations.Test
@CompileStatic
@Test
class FooTest {
    void test() {
        assert true
    }
}
```

`build.gradle.kts`
```
plugins {
    groovy
}
repositories {
    jcenter()
}
dependencies {
    implementation("org.codehaus.groovy:groovy-all:3.0.5")
    testImplementation("org.testng:testng:7.3.0")
}
tasks.test {
    useTestNG()
}
```

Observations:

1. No issue with groovy version 2.x

2. TestNG selects `setMetaClass` and `setProperty` as test methods.
When runs with ReportNG:
```
org.testng.TestNGException:
Cannot inject @Test annotated Method [setMetaClass] with [interface groovy.lang.MetaClass].
```
```
org.testng.TestNGException:
Cannot inject @Test annotated Method [setProperty] with [class java.lang.String, class java.lang.Object].
```

### PR DESCRIPTION
**Pull request by ispringer**
Fixes #2360

### Did you remember to?

- [x] Add test case(s)
- [x] Update `CHANGES.txt`

### CODE REVIEW

**Comment by ispringer** [comment]
Please hold off on merging this until I get approval from my employer (Salesforce). Hopefully the process will only take a few days. Thanks!

**Comment by ispringer** [comment]
Ok, I got approval! Merge away if it looks good.

**Review by krmahadevan — APPROVED** [review]

**Comment by juherr** [comment]
Thanks for the fix 👍

**Pull request merged**
